homeassistant/config_entries.py

```diff
@@ -3325,6 +3396,70 @@ def async_create_entry(
 
         return result
 
+    @callback
+    def async_update_and_abort(
+        self,
+        entry: ConfigEntry,
+        subentry: ConfigSubentry,
+        *,
+        unique_id: str | None | UndefinedType = UNDEFINED,
+        title: str | UndefinedType = UNDEFINED,
+        data: Mapping[str, Any] | UndefinedType = UNDEFINED,
+        data_updates: Mapping[str, Any] | UndefinedType = UNDEFINED,
+    ) -> SubentryFlowResult:
```

Comment:
The corresponding method for config entries is called `ConfigFlow.async_update_reload_and_abort` and as the name suggests also aborts.
Maybe this method should be renamed to `ConfigSubentryFlow.async_update_reload_and_abort` and reload the config entry so it behaves more similar?

How come the helper method `ConfigFlow.async_update_reload_and_abort` reloads, do we no longer expect integrations to implement their own config entry update listener?

Reply:
Actually I think that is a bug (or at least not considered when it was created).
If the update listener is reloading the integration, it does so twice now if you have this function in place.

We need the update listener for option flows, and probably having this helper in place makes sense (as lots of integrations doesn't implement an update listener as a result of having it). Should we cancel the update listener before updating the entry (as it gets reloaded a few lines down)?

Reply:
When a subentry is added should be able reload the config entry.  When a handler is added on subentry changes it could be possible to handle subentry changes without reloading the parent config entry. But may be I misunderstand?

Reply:
> When a handler is added on subentry changes it could be possible to handle subentry changes without reloading the parent config entry

No, you're right, an integration could handle it like that. Let's just add the non-reloading helper method for now then, we can always add a reload version of it later.

Reply:
See https://github.com/home-assistant/core/pull/133450 for context. I think perhaps the reloading should not have been added in the first place.